packages/server/src/client.ts

```diff
@@ -229,6 +230,7 @@ export class OpenFeatureClient implements Client {
         errorMessage,
         value: defaultValue,
         reason: StandardResolutionReasons.ERROR,
+        flagMetadata: Object.freeze({}),
```

Comment:
I'm OK with this... I think it's better than keeping it undefined. It's a bit weird that this field is here and "always empty" in error cases. I would be nice if we could mark it undefined in the type if the `reason` is error.

On the other hand, that typing wouldn't help a plain JS user, and they could get into trouble.

Reply:
Yeah thats what I thought.
Question to me was if read-only is good enough or if we need to have the promise by freeze.

Reply:
I don't have a strong opinion either way, honesty. I'll see if I can create a discriminated union type that doesn't include flag metadata when the reason is error.

Reply:
> I don't have a strong opinion either way, honesty. I'll see if I can create a discriminated union type that doesn't include flag metadata when the reason is error.

If you can, great. I dont think there's much risk here either way TBH since this is just empty in the error case. I think it's fine to keep it as is.

Reply:
I couldn't find a way to do this without massively complicating the type. For the sake of maintainability, I'll leave it as is.